cpp/source/deal.II/geometry.h

```diff
@@ -8,104 +8,72 @@
 #ifndef CAP_GEOMETRY_H
 #define CAP_GEOMETRY_H
 
+#include <cap/types.h>
 #include <deal.II/base/types.h>
-#include <deal.II/grid/tria.h>
+#include <deal.II/distributed/tria.h>
+#include <boost/mpi.hpp>
 #include <boost/property_tree/ptree.hpp>
 #include <memory>
 #include <unordered_map>
 
 namespace cap
 {
 /**
- * Base class for the geometry. This class allows access to the underlying
+ * This class allows access to the underlying
  * triangulation and the material id map. It also creates the triangulation by
  * reading the mesh from a ucd file.
  */
 template <int dim>
 class Geometry
 {
 public:
-  Geometry(std::shared_ptr<boost::property_tree::ptree const> const &database);
+  /**
+   * This contructor uses a mesh in ucd format. The database is used to get the
+   * name of the mesh file and to create the materials map.
+   */
+  Geometry(std::shared_ptr<boost::property_tree::ptree const> database,
+           boost::mpi::communicator mpi_communicator);
+
+  /**
+   * This contructor uses the dealii::distributed::Triangulation @p
+   * triangulation. If the database is necessary to create the materials map.
+   */
+  Geometry(
+      std::shared_ptr<boost::property_tree::ptree const> database,
+      std::shared_ptr<dealii::distributed::Triangulation<dim>> triangulation);
```

Comment:
I just want to be sure that we are using the same communicator everywhere. I know it caused a lot of problems in deal when people use multiple communicator.